### ISSUE TITLE
The freezing BN setting during training.

### ISSUE THREAD

**Issue by Leno-B** [BIT-DA/GDCAN#5]
Hello, thank you for the excellent job and the code. However, I'm confused about the training setting in train_dcan.py.

`        # freeze BN layers           
        for m in base_network.modules():                  
            if isinstance(m, nn.BatchNorm2d):                
                m.training = False                 
                m.weight.requires_grad = False              
                m.bias.requires_grad = False                  
`

Why freeze the BN layers during training?

**Comment by BinhuiXie** [comment]
Hi @Leno-B, it's a common practice to freeze BN while training models. This is done because you usually start from some ImageNet pre-trained weights which were trained with large batch size (like 256 or larger), while in domain adaptation the batch size per domain is much smaller. You don't want to ruin good pre-trained statistics so freezing BN is a good idea.

**Comment by gbliao** [comment]
Thanks for your prompt response. Besides, during the pretraining parameters loading process, I found the pretrained_dicts of dcca_resnet50 are loaded from attention_resnet50. 
(1) What is the attention_resnet50 network? ResNet+SE?
(2) What is the performance of dcca_resnet50 without loading the pretrained attention_resnet50? Is it the key for improving the performance of dcca_resnet50?

**Comment by gbliao** [comment]
Thanks for your prompt response. Besides, during the pretraining parameters loading process, I found the pretrained_dicts of dcca_resnet50 are loaded from attention_resnet50. 
(1) What is the attention_resnet50 network? ResNet+SE?
(2) What is the performance of dcca_resnet50 without loading the pretrained attention_resnet50? Is it the key for improving the performance of dcca_resnet50?

**Comment by Leno-B** [comment]
> Hi @Leno-B, it's a common practice to freeze BN while training models. This is done because you usually start from some ImageNet pre-trained weights which were trained with large batch size (like 256 or larger), while in domain adaptation the batch size per domain is much smaller. You don't want to ruin good pre-trained statistics so freezing BN is a good idea.

Thanks for your prompt response. 
Besides, during the pretraining parameters loading process, I found the pretrained_dicts of dcca_resnet50 are loaded from attention_resnet50.
(1) What is the attention_resnet50 network? ResNet+SE?
(2) What is the performance of dcca_resnet50 without loading the pretrained attention_resnet50? Is it the key to improve the performance of dcca_resnet50?
